src/Facebook/Facebook.php

```diff
@@ -327,7 +350,7 @@ public function getRedirectLoginHelper()
    */
   public function getJavaScriptHelper()
   {
-    return new FacebookJavaScriptHelper($this->app);
+    return new FacebookJavaScriptHelper($this->app, $this->getOAuth2Client());
```

Comment:
Still at work, but here a quick solution :)

As the OAuth2 client is used as an internat instance, with helper methods, I think the app should be the first parameters of each helper method like `getAuthUrl()` or `getATFromCode()`.
In fact, with the helper OAuth client, we can get an auth url for any app, so it should be a parameter.